roles/inject_openshift_kube_objects/tasks/main.yml

```diff
@@ -1,29 +1,34 @@
 ---
-- name: "Create the kube_objects directory"
-  shell: "mkdir -p {{ kube_objects_dir }}"
+- name: "Copy the gpg key to a file"
+  copy:
+    content: "{{ gpg_private_key }}"
+    dest: "{{ lookup('env', 'HOME') }}/gpg_private_key"
+    mode: 0600
+
+- name: "Copy the passphrase for the gpg key to a file"
+  copy:
+    content: "{{ gpg_passphrase }}"
+    dest: "{{ lookup('env', 'HOME') }}/gpg_private_key_passphrase"
+    mode: 0600
+
+- name: "Import the gpg key"
+  shell: "gpg --passphrase-file {{ lookup('env', 'HOME') }}/gpg_private_key_passphrase --pinentry-mode loopback --import {{ lookup('env', 'HOME') }}/gpg_private_key"
+  register: result
+  failed_when:
+    - result.rc != 2
 
-- name: "Decode the symmetric key received from the user"
-  shell: "echo \"{{ symmetric_key }}\" | base64 --decode > {{ kube_objects_dir }}/get_sym_key"
-  no_log: true
-  
-- name: "Decrypt the symmetric key using RSA private key"
-  shell: "openssl rsautl -decrypt -oaep -inkey {{ rsa_private_key }} -in {{ kube_objects_dir }}/get_sym_key -out {{ kube_objects_dir }}/secret.key"
+- name: "Create the kube_objects directory"
+  file:
+    path: "{{ kube_objects_dir }}"
+    state: directory
+    mode: '0755'
 
-- name: "Verify the shared symmetric key"
-  shell: "stat -c %s {{ kube_objects_dir }}/secret.key"
-  register: symmetric_key_size
 
-- name: "Fail if the shared symmetric key is not 32 bytes"
-  fail:
-    msg: "The decrypted shared symmetric key is {{ symmetric_key_size.stdout }} bytes. The shared key should be of 32 bytes"
-  when: symmetric_key_size.stdout != "32"
+- name: "Read the encrypted kube_objects, decode and store them in a file"
+  shell: "echo \"{{ kube_objects }}\" > {{ kube_objects_dir }}/get_kubeObjects.txt"
 
-- name: "Read the encrypted kube_objects decode and store them in a file"
-  shell: "echo \"{{ kube_objects }}\" | tr \" \" \"\n\" | base64 --decode > {{ kube_objects_dir }}/get_kubeObjects.txt"
-  no_log: true
-  
-- name: "Decrypt the kube_objects through the shared symmetric key"
-  shell: "openssl aes-256-cbc -d -in {{ kube_objects_dir }}/get_kubeObjects.txt -out {{ kube_objects_dir }}/kube_objects.yaml -pass file:{{ kube_objects_dir }}/secret.key"
+- name: "Decrypt the kube_objects using the gpg key"
+  shell: "gpg --decrypt --pinentry-mode loopback --passphrase-file {{ lookup('env', 'HOME') }}/gpg_private_key_passphrase {{ kube_objects_dir }}/get_kubeObjects.txt > {{ kube_objects_dir }}/kube_objects.yaml"
```

Comment:
Have you consider to run gpg-agent ?
it will allow to move secrets out of playbook and decryption will be just simpler one-liner
e.g.
without gpg agent
```
[cvp-ops@5357f25e051b ~]$ gpg --output ./doc --decrypt doc.gpg
gpg: encrypted with 4096-bit RSA key, ID xxxxxxxxxxxxxx, created 2020-05-19
      "Container Verification Pipeline (Jenkins Pipeline key) <xxxxx@redhat.com>"
gpg: public key decryption failed: Permission denied
gpg: decryption failed: No secret key
```

with gpg-agent
```
[cvp-ops@5357f25e051b ~]$ gpg-agent --daemon --allow-preset-passphrase
[cvp-ops@5357f25e051b ~]$ /usr/libexec/gpg-preset-passphrase --preset --passphrase XXXXXXXXXX $KEYGRIP
[cvp-ops@5357f25e051b ~]$ rm -f doc
[cvp-ops@5357f25e051b ~]$ gpg --output ./doc --decrypt doc.gpg
gpg: encrypted with 4096-bit RSA key, ID 3F781BD452E59E04, created 2020-05-19
      "Container Verification Pipeline (Jenkins Pipeline key) <cvp-ops@redhat.com>"
[cvp-ops@5357f25e051b ~]$ echo $?
0
[cvp-ops@5357f25e051b ~]$ ls -l doc
-rw-rw-r--. 1 cvp-ops cvp-ops 175 Oct  1 12:03 doc
```

Reply:
What does gpg agent do that I am not already doing? 

The user of this playbook uses gpg-agent to input the keys and passphrase, and then the handling of those doesn't have to be done in the playbook?

Reply:
Also, how does gpg-agent import the key? I don't see an import step there, just setting a passphrase.

Reply:
Oh I see your MR in CVP now, so there is an extra step that is missing. So whoever is using these playbooks has to run gpg-agent and then `gpg --batch --import `?

Reply:
The idea of using gpg-agent is to avoid passing private data to the playbook.
Because if you want to hide secrets in playbook you need to use `no_log: true` which complicates debugging
and we do not want to leak our secrets